Shared/Samples/Find route around barriers/FindRouteAroundBarriersView.Views.swift

```diff
@@ -16,43 +16,47 @@ import ArcGIS
 import SwiftUI
 
 extension FindRouteAroundBarriersView {
-    /// The list of settings for the sample.
-    struct SettingsList: View {
-        /// The view model for the sample.
-        @ObservedObject var model: Model
+    /// A list of settings for modifying route parameters.
+    struct RouteParametersSettingsList: View {
+        /// The route parameters to modify.
+        private let routeParameters: RouteParameters
         
         /// A Boolean value indicating whether routing will find the best sequence.
-        @State private var routingFindsBestSequence = false
+        @State private var routingFindsBestSequence: Bool
+        
+        /// A Boolean value indicating whether routing will preserve the first stop.
+        @State private var routePreservesFirstStop: Bool
+        
+        /// A Boolean value indicating whether routing will preserve the last stop.
+        @State private var routePreservesLastStop: Bool
+        
+        init(for routeParameters: RouteParameters) {
+            self.routeParameters = routeParameters
+            _routingFindsBestSequence = State(initialValue: routeParameters.findsBestSequence)
+            _routePreservesFirstStop = State(initialValue: routeParameters.preservesFirstStop)
+            _routePreservesLastStop = State(initialValue: routeParameters.preservesLastStop)
```

Comment:
Any specific reason to use `State(initialValue:)`?
```suggestion
            self.routingFindsBestSequence = routeParameters.findsBestSequence
            self.routePreservesFirstStop = routeParameters.preservesFirstStop
            self.routePreservesLastStop = routeParameters.preservesLastStop

```

Reply:
[The doc](https://developer.apple.com/documentation/swiftui/state/init(wrappedvalue:)) is pretty clear that state properties should not be set in an initializer, neither by explicitly calling `init(initialValue:)` nor by implicitly calling `init(wrappedValue:)`:

> You don’t call this initializer directly. Instead, SwiftUI calls it for you when you declare a property with the @State attribute and provide an initial value

The reason being is that state is a private, implementation detail. The parent view's body being re-computed shouldn't alter the view's state, which is what happens when a state property is set from inside an initializer. For values provided by an initializer, non-state properties should be used.

Reply:
@philium sorry I missed your comment. Would it be better to use bindings here in this case?

Reply:
I took a closer look at what this sample is doing and in this case, I don't see an alternative option that isn't more complicated. As I mentioned, the concern with setting the state in an initializer is that the state will be reset every time the parent's body is re-computed, but in this case, that isn't an issue. (In an ideal world, `RouteParameters` would be a struct and the settings view would take a binding to it. Then everything would just work without needing separate state properties.)

A couple other things I noticed:

- The settings view initializer parameter label should be `routeParameters` instead of `for`. From the [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/): "The first argument to initializer…calls should not form a phrase starting with the base name, e.g. x.makeWidget(cogCount: 47)"
- The settings view should use `Form` instead of `List`, as `Form` is "a container for grouping controls used for data entry, such as in settings or inspectors."

Reply:
@philium Sounds good. I'll make another PR to address those issues.

Reply:
> The settings view should use Form instead of List

here is a [card](https://github.com/Esri/arcgis-maps-sdk-swift-samples/projects/1#card-89771848) for SprintX 😼